scripts/launch/Test.hs

```diff
@@ -0,0 +1,214 @@
+bvMinorL = bvL . key "bvMinor" . _Number
+bvMajorL = bvL . key "bvMajor" . _Number
+
+updateTest :: Y.Value -> Config -> IO ()
+updateTest config (Config{..}) = do
+    let config' = config & appNameL .~ walletAppName
+        config'' = config' & svL %~ (+1)
+        showF = T.pack . show @Int . round
+        cFile' = T.pack $ F.encodeString cFile
+        (proposeUpdStr, cNodes) = fromMaybe (error "Failed to form update proposal cmd") $ do
+            tag <- config ^? tagL
+            bvAlt <- config ^? bvAltL
+            bvMinor <- config ^? bvMinorL
+            bvMajor <- config ^? bvMajorL
+            genSlot <- config ^? genSlotL
+            genScriptVersion <- config ^? genScriptVersionL
+            genBlockSize <- config ^? genBlockSizeL
+            sv <- config ^? svL
+            (cNodes :: Int) <- round <$> config ^? richL
+            let bvmStr = "bvm"
+                    <> " script-version: " <> showF genScriptVersion
+                    <> " slot-duration: " <> showF (genSlot / 1000)
+                    <> " max-block-size: " <> showF genBlockSize
+                updStr = "upd-bin"
+                    <> " system: " <> T.pack (show tag)
+                    <> " installer-path: ./test_updater.sh"
+            return $ (,cNodes) $ "propose-update i:0"
+              <> " block-version: " <> showF bvMajor <> "." <> showF bvMinor <> "." <> showF bvAlt
+              <> " software-version: ~software~" <> walletAppName <> ":" <> showF (sv + 1)
+              <> " bvm: (" <> bvmStr <> ")"
+              <> " update: (" <> updStr <> ")"
+              <> " vote-all: true"
+    TIO.putStrLn proposeUpdStr
+    Y.encodeFile "run/conf-fresh.yaml" (M.singleton cKey config'')
+
+    TIO.writeFile "test_updater.sh" $ "cp run/conf-fresh.yaml run/configuration.wallet.yaml"
+    let cmdPropose = T.intercalate ";"
+            [ "add-key-pool " <> T.unwords (format ("i:"%d) <$> [0..cNodes-1])
+            , "listaddr"
+            , proposeUpdStr
+            ]
+        auxx cmds = ("cardano-auxx",)
+            [ "--keyfile", "run/auxx.keys"
+            , "--configuration-key", cKey
+            , "--configuration-file", cFile'
+            , "--system-start", "0"
+            , "--peer", "127.0.0.1:3001"
+            , "--db-path", "run/auxx-db"
+            , "--log-config", "scripts/log-templates/log-config-greppable.yaml"
+            , "cmd", "--commands", cmds
+            ]
+    (exitCode, textToLines -> auxxOut) <- uncurry procStrict (auxx cmdPropose) empty
+    case exitCode of
+        ExitSuccess -> do
+            echo "Proposing an update succeeded, here are the logs:"
+            mapM_ (echo . mappend "\t") auxxOut
+        ExitFailure _ -> do
+            echo "Proposing an update failed, see auxx output:"
+            mapM_ (echo . mappend "\t") auxxOut
+            exit exitCode
+    let matchInAuxxOut pat = do
+            (lineToText -> t) <- NE.toList auxxOut
+            match (suffix pat) t
```

Comment:
I think that in this case it would've been sli-ighly better to avoid ViewPatterns.

```haskell
t <- NE.toList auxxOut
match (suffix pat) (lineToText t)
```